File: backend/app/schemas/evaluation.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.project_candidate import ProjectCandidateStatus
from app.schemas.candidate import CandidateResponse


class ProjectCandidateResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    candidate_id: uuid.UUID
    match_score: float | None
    dimension_scores: dict | None
    recommendation: str | None
    risk_flags: str | None
    hunter_feedback: str | None
    status: ProjectCandidateStatus
    evaluated_at: datetime | None
    candidate: CandidateResponse | None = None


class ProjectCandidateUpdate(BaseModel):
    status: ProjectCandidateStatus | None = None
    hunter_feedback: str | None = None


class EvaluationStatusResponse(BaseModel):
    project_id: uuid.UUID
    candidate_id: uuid.UUID
    is_complete: bool               # True once EA has finished and evaluated_at is set
    status: ProjectCandidateStatus
    match_score: float | None       # null until evaluation completes
    evaluated_at: datetime | None   # null until evaluation completes
    poll_interval_seconds: int = 5  # recommended polling interval

File: frontend/src/types/evaluation.ts
import type { CandidateResponse } from './candidate'

export type ProjectCandidateStatus = 'pending' | 'recommended' | 'eliminated' | 'interviewed'

export interface ProjectCandidateResponse {
  id: string
  project_id: string
  candidate_id: string
  match_score: number | null
  dimension_scores: Record<string, number> | null
  recommendation: string | null
  risk_flags: string | null
  hunter_feedback: string | null
  status: ProjectCandidateStatus
  evaluated_at: string | null
  trigger_source: string | null
  llm_raw_output: string | null
  candidate: CandidateResponse | null
}

export interface CandidateEvaluationResponse {
  id: string
  project_id: string
  project_name: string
  client_name: string
  match_score: number | null
  recommendation: string | null
  risk_flags: string | null
  trigger_source: string | null
  status: ProjectCandidateStatus
  evaluated_at: string | null
}

export interface ProjectCandidateUpdate {
  status?: ProjectCandidateStatus | null
  hunter_feedback?: string | null
}

export interface EvaluationStatusResponse {
  project_id: string
  candidate_id: string
  is_complete: boolean
  status: ProjectCandidateStatus
  match_score: number | null
  evaluated_at: string | null
  poll_interval_seconds: number
}
